common/tmx_filetype.go

```diff
@@ -0,0 +1,60 @@
+package common
+
+import (
+	"bytes"
+	"fmt"
+        "engo.io/engo"
+	"io"
+)
+
+// TMXResource is a wrapper for a level that was created from a Tile Map XML
+type TMXResource struct {
+	Level *Level
+	url   string
+}
+
+func (r TMXResource) URL() string {
+	return r.url
+}
+
+// tmxLoader is responsible for managing '.tmx' files within 'engo.Files'
+type tmxLoader struct {
+	tmxs map[string]TMXResource
+}
+
+// Load will read the tmx file into a string to create a level using createLevelFromTmx.  Any image files required for creating the level will be loaded in as needed.
```

Comment:
Split this into multiple lines and I don't think you need to mention createLevelFromTmx